### PR TITLE
Problem: Cluster DNS and Health check terminology is incorrect

### LINKED ISSUE #2140: Problem: Cluster DNS and Health check terminology is incorrect
* BigchainDB version: 2.0
* Operating System: N/A
* Deployment Type: `Kubernetes`
* BigchainDB driver: `no`

### Description
The following configurations i.e. `cluster-health-check-port` and `cluster-dns-server-ip` were incorrect because they are only specific to a single BigchainDB node, replace them with `node-health-check-port` and `node-dns-server-ip`. This also impacts documentation.

Currently, the `cluster-dns-server-ip` is hardcoded in helper automation functions but it is not the same for all Kubernetes deployments. Make it configurable.

### PR DESCRIPTION
**Pull request by muawiakh**
## Description
The following configurations i.e. `cluster-health-check-port` and `cluster-dns-server-ip` were incorrect because they are only specific to a single BigchainDB node hence replacing them with `node-health-check-port` and `node-dns-server-ip`. Replacing all the occurrences in the documentation.

Currently, the `cluster-dns-server-ip` was hardcoded but it is not the same for all Kubernetes deployments. Made it configurable and added documentation.

Also changed `bdb-cluster-ca` naming to `bdb-node-ca`.

## Issues This PR Fixes
Fixes #2140

### CODE REVIEW

File: docs/server/source/production-deployment-template/node-on-kubernetes.rst [hunk 1]
```diff
@@ -73,7 +73,7 @@ to the above command (i.e. the path to the private key).
       $ kubectl get pods
 
     will get a list of the pods in the Kubernetes cluster associated
-    with the context named ``k8s-bdb-test-cluster-0``.
+    with the context named ``k8s-bdb-test-node-0``.
 
 Step 2: Connect to Your Cluster's Web UI (Optional)
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/node-on-kubernetes.rst]: Should this say "Your Cluster's Web UI" or "Your Node's Web UI"? Technically, it _is_ a Kubernetes cluster (but the cluster is just for one node)...

File: k8s/configuration/config-map.yaml [hunk 2]
```diff
@@ -14,15 +14,15 @@ data:
...
-  cluster-health-check-port: "8888"
+  node-health-check-port: "8888"
 
-  # cluster-dns-server-ip is the IP of the DNS server. A Kubernetes deployment
+  # node-dns-server-ip is the IP of the DNS server. A Kubernetes deployment
   # always has a DNS server (kube-dns) running at 10.0.0.10
-  cluster-dns-server-ip: "10.0.0.10"
+  node-dns-server-ip: "10.0.0.10"
 
   # mdb-instance-name is the name of the MongoDB instance in this cluster.
```
Comment (ttmc) [inline k8s/configuration/config-map.yaml]: "...in this cluster" or "...in this node"?

File: k8s/scripts/vars [hunk 3]
```diff
@@ -38,4 +38,9 @@ TM_GENESIS_TIME='0001-01-01T00:00:00Z'
 
 # Blockchain ID must be unique for
 # every blockchain
-TM_CHAIN_ID='test-chain-rwcPML'
\ No newline at end of file
+TM_CHAIN_ID='test-chain-rwcPML'
+
+# IP Address of the resolver(DNS server).
+# i.e. CLUSTER-IP of `kubernetes` service(kube-dns)
```
Comment (ttmc) [inline k8s/scripts/vars]: CLUSTER-IP or NODE-IP?

File: docs/server/source/production-deployment-template/node-config-map-and-secrets.rst [hunk 4]
```diff
@@ -69,6 +69,19 @@ These parameters are shared across the cluster. More information about the gener
 of these parameters can be found at :ref:`generate-the-blockchain-id-and-genesis-time`.
 
 
+vars.NODE_DNS_SERVER
+^^^^^^^^^^^^^^^^^^^^
+`CLUSTER-IP` of Kubernetes service(kube-dns), can be retrieved using
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/node-config-map-and-secrets.rst]: CLUSTER-IP or NODE-IP?

File: k8s/scripts/functions [hunk 5]
```diff
@@ -242,15 +243,15 @@ data:
...
+  node-health-check-port: "8888"
 
-  # cluster-dns-server-ip is the IP of the DNS server. A Kubernetes deployment
-  # always has a DNS server (kube-dns) running at 10.0.0.10
-  cluster-dns-server-ip: "10.0.0.10"
+  # node-dns-server-ip is the IP of the DNS server. A Kubernetes deployment
+  # always has a DNS server (kube-dns). 
+  node-dns-server-ip: "${dns_resolver_k8s}"
 
   # mdb-instance-name is the name of the MongoDB instance in this cluster.
```
Comment (ttmc) [inline k8s/scripts/functions]: Maybe change to "...in this node."

File: docs/server/source/production-deployment-template/node-on-kubernetes.rst (hunk 1 above)
Comment (muawiakh) [reply docs/server/source/production-deployment-template/node-on-kubernetes.rst]: Agreed.

File: k8s/configuration/config-map.yaml [hunk 6]
```diff
@@ -14,15 +14,15 @@ data:
...
   # In our deployment, Kubernetes sends 'livenessProbes' to this port and
   # interprets a successful response as a 'healthy' service.
-  cluster-health-check-port: "8888"
+  node-health-check-port: "8888"
 
-  # cluster-dns-server-ip is the IP of the DNS server. A Kubernetes deployment
+  # node-dns-server-ip is the IP of the DNS server. A Kubernetes deployment
   # always has a DNS server (kube-dns) running at 10.0.0.10
-  cluster-dns-server-ip: "10.0.0.10"
+  node-dns-server-ip: "10.0.0.10"
```
[the remark this answers is not in the document]
Comment (muawiakh) [reply k8s/configuration/config-map.yaml]: Makes sense.

File: k8s/configuration/config-map.yaml (hunk 2 above)
Comment (muawiakh) [reply k8s/configuration/config-map.yaml]: cluster because it refers to Kubernetes.

File: k8s/scripts/vars (hunk 3 above)
Comment (muawiakh) [reply k8s/scripts/vars]: `CLUSTER-IP` naming is a bit confusing but Kubernetes calls that IP `CLUSTER-IP`

File: docs/server/source/production-deployment-template/node-config-map-and-secrets.rst (hunk 4 above)
Comment (muawiakh) [reply docs/server/source/production-deployment-template/node-config-map-and-secrets.rst]: Same as above.

File: k8s/scripts/functions (hunk 5 above)
Comment (muawiakh) [reply k8s/scripts/functions]: Node would make sense too.

File: k8s/scripts/vars (hunk 3 above)
Comment (muawiakh) [reply k8s/scripts/vars]: I will rename to `IP` to remove the confusion.

**Comment by ttmc** [comment]
I think you intended to replace

`Your Cluster's Web UI`

with

`Your Kubernetes Cluster's Web UI`

e.g. in `node-on-kubernetes.rst` on line 78.

**Pull request merged**